### PR TITLE
Use fixed container URI 

### LINKED ISSUE #209: Problems without specific revision of pipeline when pulling via Singularity
Hi!
neither on local nor on our cluster (binac) I can download the appropriate image... 
```
[warm up] executor > local
ERROR ~ Error executing process > 'fastqc (null)'

Caused by:
  Failed to pull singularity image
  command: singularity pull --name scilifelab-ngi-rnaseq-master.img docker://scilifelab/ngi-rnaseq:master > /dev/null
  status : 1
  message:
    WARNING: pull for Docker Hub is not guaranteed to produce the
    WARNING: same image on repeated pull. Use Singularity Registry
    WARNING: (shub://) to pull exactly equivalent images.
    ERROR MANIFEST_UNKNOWN: manifest unknown
    ERROR: pulling container failed!
-- Check '.nextflow.log' file for details
```

I think the problematic part is this code piece here in `base.config`:

```nextflow
container = { "scilifelab/ngi-rnaseq:${workflow.revision ? workflow.revision : 'latest'}" }
``` 

This defaults to something else, namely namely this command:

``` 
singularity pull --name scilifelab-ngi-rnaseq-master.img docker://scilifelab/ngi-rnaseq:master
```
Which doesn't work unfortunately - removing the master and things are fine ;-) 

I can send a pull request to fix this in case you'd be interested.

### PR DESCRIPTION
**Pull request by apeltzer**
This should fix #209 , and we should be able to set this to a fixed value when publishing a new stable version of the pipeline at any point using the suggestions Phil provided in #209.

### CODE REVIEW

**Comment by ewels** [comment]
Great, thanks! This could be a nice opportunity to follow the new nf-core guidelines if possible - move `main.nf / version` to `nextflow.config / params.version` https://github.com/SciLifeLab/NGI-RNAseq/blob/263fb30d097d61c12920b9c82e34c19bea0775cc/main.nf#L77

Then we can define both `params.version` and `container` as variables in `nextflow.config` next to one another so that it's easy to see that they should be related. Then the `base.config` / any other configs can pull in the container name from this variable. What do you think?

**Comment by apeltzer** [comment]
Absolutely - will change this accordingly and sync afterwards!

File: nextflow.config [hunk 1]
```diff
@@ -15,6 +15,15 @@ params {
...
   clusterOptions = false
+  
+  // Pipeline version
+  version = '1.4dev'
+  /*
+  * Variable to specify the docker / singularity image tag to use
+  * Picks up on use of -r 1.3 in nextflow command
+  * Stable releases use a tagged version of the container (e.g. scilifelab/ngi-rnaseq:1.5) whereas dev versions use the most up to date version of the container (see issue #209 on github for details).
+  */
+  container = { "scilifelab/ngi-rnaseq" }
```
Comment (ewels) [inline nextflow.config]: This is now `params.container` instead of `process.container`, so will break I think.

File: nextflow.config [hunk 2]
```diff
@@ -15,6 +15,15 @@ params {
   outdir = './results'
   igenomes_base = 's3://ngi-igenomes/igenomes/'
   clusterOptions = false
+  
+  // Pipeline version
+  version = '1.4dev'
```
Comment (ewels) [inline nextflow.config]: This is good, but the variable name in `main.nf` now needs to be updated to `params.version` instead of just `version`.

Comment (ewels) [reply nextflow.config]: Also, small thing but if these two lines could be at the top that would be great, as they're the only parts of this config that need to be regularly updated.

File: nextflow.config [hunk 3]
```diff
@@ -15,6 +15,15 @@ params {
...
+  /*
+  * Variable to specify the docker / singularity image tag to use
+  * Picks up on use of -r 1.3 in nextflow command
```
Comment (ewels) [inline nextflow.config]: This is not true any more..

File: conf/base.config [hunk 4]
```diff
@@ -12,10 +12,8 @@ vim: syntax=groovy
  */
 
 process {
-
-  // Variable to specify the docker / singularity image tag to use
-  // Picks up on use of -r 1.3 in nextflow command
-  container = { "scilifelab/ngi-rnaseq:${workflow.revision ? workflow.revision : 'latest'}" }
```
Comment (ewels) [inline conf/base.config]: We need to leave this here as we need `process.container`. Can just set it to `params.container` to pick up on the variable that's been defined in `nextflow.config`.

**Comment by ewels** [comment]
Corresponding fix done in nf-core/methylseq#8

**Comment by ewels** [comment]
Nearly there - just missing two occurrences of `version` in `main.nf`:

* https://github.com/apeltzer/NGI-RNAseq/blob/fix-container-reference/main.nf#L22
* https://github.com/apeltzer/NGI-RNAseq/blob/fix-container-reference/main.nf#L1136

**Comment by ewels** [comment]
Awesome stuff, thanks @apeltzer! 🎉

**Comment by ewels** [comment]
Hmm, looks like I missed an error in the travis build: https://travis-ci.org/SciLifeLab/NGI-RNAseq/jobs/355755444#L716

```
ERROR ~ Failed to invoke `workflow.onComplete` event handler
 -- Check script 'main.nf' at line: 1132 or see '.nextflow.log' file for more details
```

It's this line:

```nextflow
email_fields['version'] = $params.version
```

I think it should be `params.version` (without the `$`)...

**Pull request merged**